src/main/java/de/uniwue/web/controller/LibraryController.java

```diff
@@ -161,21 +161,45 @@ public String home(Model model) throws IOException {
 	}
 	/**
 	 * returns each imagePath in given directory
+	 * respecting the SubExtensionFilter set in properties
 	 *
 	 * @param baseFolder path to legacy baseFolder
 	 * @param extList file extension to map
 	 * @return map containing imageName and path
 	 */
-	public Map<String, String> getFileMap(String baseFolder, List<String> extList) {
-		Map<String, String> fileMap = new LinkedHashMap<String, String>();
+	public Map<String, List<String>> getFileMap(String baseFolder, List<String> extList) {
+		//Moved imageSubFilter processing here
+		List<String> imageSubFilter = config.getListSetting("imagefilter");
+		if(!imageSubFilter.isEmpty()) {
+			List<String> extListWithSub = new ArrayList<>();
+			for(String ext : extList) {
+				for(String subExt : imageSubFilter)
+				extListWithSub.add("." + subExt + ext);
+			}
+			extList = extListWithSub;
+		}
+		Map<String, List<String>> fileMap = new TreeMap<>();
 		File directFolder = new File(baseFolder);
-		List<File> files = Arrays.stream(directFolder.listFiles()).sorted(Comparator.naturalOrder()).collect(Collectors.toList());
+		File[] files = directFolder.listFiles();
+
 		for (File file : files) {
+			List<String> images = new ArrayList<>();
+			String currentFileName = file.getName().split("\\.")[0];
 			for (String ext : extList) {
 				if(file.getName().endsWith(ext)) {
 					String path = file.getAbsolutePath();
-					fileMap.put(file.getName(), path);
+					images.add(path);
+				}
+			}
+			if(!images.isEmpty()) {
+				if(fileMap.containsKey(currentFileName)) {
+					images.addAll(fileMap.get(currentFileName));
+					fileMap.put(currentFileName,images);
+				} else {
+					fileMap.put(currentFileName,images);
 				}
+				images.sort(Comparator.naturalOrder());
+				fileMap.put(currentFileName,images);

```

Comment:
`fileMap.put(currentFileName,images);` will get run in all cases so it can be extracted and `else` removed:
```Java
if(fileMap.containsKey(currentFileName)) {
	images.addAll(fileMap.get(currentFileName));
}
fileMap.put(currentFileName,images);
images.sort(Comparator.naturalOrder());
fileMap.put(currentFileName,images);
```

Reply:
removed with [c296372 ](https://github.com/OCR4all/LAREX/pull/273/commits/c296372516aa9146ecaf5f7f8d27d9454516bca4)